pytorch_lightning/loggers/base.py

```diff
@@ -102,7 +102,24 @@ def _reduce_agg_metrics(self):
         elif len(self._metrics_to_agg) == 1:
             agg_mets = self._metrics_to_agg[0]
         else:
-            agg_mets = merge_dicts(self._metrics_to_agg, self._agg_key_funcs, self._agg_default_func)
+            # check if dictionary keys are unique
+            agg_keys = set([key for met in self._metrics_to_agg for key in met.keys()])
+            num_keys = sum([len(met) for met in self._metrics_to_agg])
+
+            # exclude 'epoch' because it is a metric automatically added in by log_metrics and will count as a
+            # duplicate. If you want to get rid of this, I would suggest you should get rid of `scalar_metrics[
```

Comment:
are the changes here really related to the global step issue?
If possible could we make a separate PR for this? we only want to fix one thing at a time.

Reply:
I can see it being a separate issue. But, this piece of code is necessary for the code to run smoothly. Otherwise, when we fix the step issue (i.e. combining the last training batch + training_epoch_end + validation_epoch_end metrics in the same step), there will be an error combining the three pieces of data together because the original `merge_dict()` in `_reduce_agg_metrics()` requires that all the metric dictionaries have the same keys - which for these three types of data is usually not the case.

I don't mind doing a separate pull request for this though - but it will probably result in most tests failing for this pull request, which we will probably have to make sure to fix in the new pull request.